packages/chain-bridge/src/router/BridgeRouter.sol

```diff
@@ -0,0 +1,829 @@
+            if (!success) revert TransferFailed();
+        }
+    }
+
+    /**
+     * @dev Internal function to generate a unique operation ID and set initial status
+     * @param operationType Type of operation being performed
+     * @param destinationChainId Target chain ID
+     * @param asset Asset address (address(0) for non-asset operations)
+     * @param amount Amount (0 for non-asset operations)
+     * @param recipient Recipient address
+     * @param additionalData Additional data for ID generation (contract address, selector, etc.)
+     * @return operationId The generated operation ID
+     */
+    function _generateOperationId(
+        BridgeTypes.OperationType operationType,
+        uint16 destinationChainId,
+        address asset,
+        uint256 amount,
+        address recipient,
+        bytes memory additionalData
+    ) internal returns (bytes32 operationId) {
+        // Use nonce for better uniqueness and collision resistance
+        uint256 currentNonce = _useNonce(address(this));
+
+        operationId = keccak256(
+            abi.encode(
+                block.chainid,
+                destinationChainId,
+                asset,
+                amount,
+                recipient,
+                additionalData,
+                currentNonce,
+                operationType
+            )
+        );
+
+        // Set initial status to QUEUED
+        operationStatuses[operationId] = BridgeTypes.OperationStatus.QUEUED;
+
+        return operationId;
+    }
+
+    /**
+     * @dev Internal function to apply fee buffer for cross-chain operation volatility
+     * @param baseFee The base fee amount to buffer
+     * @return bufferedFee The fee with 1% buffer applied
+     */
+    function _applyFeeBuffer(
+        uint256 baseFee
+    ) internal pure returns (uint256 bufferedFee) {
+        // Add 1% buffer to account for fee volatility
+        return (baseFee * 101) / 100;
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                           BRIDGE QUEUE OPERATIONS
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     * @inheritdoc IBridgeRouter
+     */
+    function executeTransferAssets(
+        BridgeTypes.ExecuteTransferParams calldata params
+    )
+        external
+        payable
+        onlyBridgeQueue
+        whenNotPaused
+        nonReentrant
+        returns (bytes32 operationId)
+    {
+        _validateTransferParams(params);
+
+        // Get required base fee and specified adapter (no multiplier)
+        (uint256 requiredBaseFee, , address specifiedAdapter) = _quote(
+            params.destinationChainId,
+            params.asset,
+            params.amount,
+            params.options,
+            BridgeTypes.OperationType.TRANSFER_ASSET
+        );
+
+        // Apply fee buffer to account for fee volatility
+        uint256 bufferedFee = _applyFeeBuffer(requiredBaseFee);
```

Comment:
since the fee is aquired offchain in the same tx, is that required ? I'm not familair with all the supported bridges